stm32: tests: drivers : fix mpu faults by enabling SRAM clocks and configuring MPU regions on stm32h7rsx

djiatsaf-st opened the pull request:
This pull request addresses multiple MPU-related faults observed during runtime tests on the `STM32H7RSX` 
series by introducing the following changes:

- Enable MPU regions for SRAM1 and SRAM2 access 
- Enables AHB2 peripheral clocks for SRAM1 and SRAM2

djiatsaf-st committed:
drivers: clock_control: stm32: enable clocks for SRAM1 and SRAM2

enables the AHB2 peripheral clocks for SRAM1 and SRAM2
on STM32H7RSX series using LL_AHB2_GRP1_EnableClock.

These clocks are required to access the corresponding SRAM regions
during runtime.

Fixes potential access faults when using SRAM1 and SRAM2.

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

erwango commented on soc/st/stm32/stm32h7rsx/mpu_regions.c:
```diff
@@ -29,6 +29,12 @@ static const struct arm_mpu_region mpu_regions[] = {
+	MPU_REGION_ENTRY("SRAM_1", 0x30000000, REGION_RAM_ATTR(REGION_16K)),
+
+	/* Region 5 */
+	MPU_REGION_ENTRY("SRAM_2", 0x30004000, REGION_RAM_ATTR(REGION_16K)),
```
Let's make use of dt attribute `zephyr,memory-attr` instead

djiatsaf-st replied:
Done thanks

erwango commented on drivers/clock_control/clock_stm32_ll_h7.c:
```diff
@@ -1040,6 +1040,10 @@ int stm32_clock_control_init(const struct device *dev)
+#if defined(CONFIG_SOC_SERIES_STM32H7RSX)
+	LL_AHB2_GRP1_EnableClock(LL_AHB2_GRP1_PERIPH_AHBSRAM1);
+	LL_AHB2_GRP1_EnableClock(LL_AHB2_GRP1_PERIPH_AHBSRAM2);
+#endif
```
Should ideally depend on RAM activation via dts (activated by default but still)

djiatsaf-st committed:
dts: arm: st: h7rs: Add zephyr,memory-attr to SRAM1 and SRAM2 regions

adds the `zephyr,memory-attr` property to the SRAM1 and SRAM2
memory nodes to explicitly define their MPU attributes as normal
RAM. This ensures proper memory protection and caching behavior
when these regions are used by the kernel or application.

Resolve a Data Access Violation encountered during
test, where the faulting address was 0x30000000.

Note: add the zephyr,memory-attr property in the board overlay for SRAM2
to avoid conflict with the support of h7rs ethernet with MPU regions
enabled.
see link below for more details :
https://github.com/zephyrproject-rtos/zephyr/pull/97364/files#r2439668915

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

djiatsaf-st commented on soc/st/stm32/stm32h7rsx/mpu_regions.c:
```diff
@@ -22,6 +22,9 @@ static const struct arm_mpu_region mpu_regions[] = {
 			 REGION_FLASH_ATTR(REGION_FLASH_SIZE)),
+
+	/* Special region */
+	MPU_REGION_ENTRY("SPECIAL_REGION", 0x08fff000, REGION_FLASH_ATTR(REGION_4K)),
```
moved

djiatsaf-st commented on dts/arm/st/h7rs/stm32h7rs.dtsi:
```diff
@@ -60,6 +60,7 @@
+		zephyr,memory-attr = <DT_MEM_ARM(ATTR_MPU_RAM)>;
 	};
 
 	/* System data RAM accessible over AHB bus: SRAM2 in D2 domain */
```
Done

djiatsaf-st commented on dts/arm/st/h7rs/stm32h7rs.dtsi:
```diff
@@ -60,13 +60,19 @@
 		reg = <0x30000000 DT_SIZE_K(16)>;
 		compatible = "zephyr,memory-region", "mmio-sram";
 		zephyr,memory-region = "SRAM1";
+		zephyr,memory-attr = <DT_MEM_ARM(ATTR_MPU_RAM)>;
```
I previously did that [https://github.com/zephyrproject-rtos/zephyr/pull/97731#discussion_r2436709223](url) but is better to do it this way unless there is a conflict, such as with SRAM2

erwango reviewed (approved):
Approved, but please update PR description:
`Configure SPECIAL_REGION to resolve access violation` doesn't apply anymore
